File: bubbleSort.cpp
#include <iostream>
using namespace std;

int* bubble_Sort(int v[],int vsize) {
    for (int i = 0; i < vsize; i++) {
        int swapped = 1;
        for (int j = 0; j < vsize - 1 - i; j++) {
            if (v[j] > v[j + 1]) {
                swap(v[j], v[j + 1]);
                swapped = 1;
            }
            if (!swapped)
                break;
        }
    }
    return v;
}
int main() {
    int v[]={1,4,2,7,8,21,2};
    int *a=bubble_Sort(v,7);
    for(int i=0;i<7;i++) {
        cout << a[i] <<" ";
    }
}

File: bubbleSort.java
package SortingAlgorithm;


public class bubbleSort {

    public static void bubble_sort(int[] a){
        int n = a.length;

        for(int i = 0; i < n - 1; i++){

            boolean flag = false;

            for(int j = 0; j < n - i - 1 ; j++) {    // time complexity = O(n^2)
                if (a[j] > a[j + 1]) {
                    int temp = a[j];
                    a[j] = a[j + 1];
                    a[j + 1] = temp;
                    flag = true;
                }
            }
            if(flag == false){ // have any swaps happened ?  yes because no swap happened
                return;
            }
        }

    }
    public static void main(String[] args) {

        int[] arr = { 5, 6, 8, 1 , 9};
        bubble_sort(arr);

        for(int i = 0; i < arr.length; i ++){
            System.out.print(arr[i] + " ");
        }

    }
}
